### PR TITLE
Documentation: Add contributing guidelines

### PR DESCRIPTION
**Pull request by tetektoza**
This patch adds contributing guidelines to the project, with guidelines on creating PRs, issues, link to the discord, and behavior etiquette.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+For questions regarding your implementation change, you can always ask on our Discord, on `#development` channel.
+
+Please try to do these when creating an issue:
+
+* Try to prepend your issue's title with the category/component that the issue is related to, for example, if your change is mostly related to LevelCelView, you'd name your issue as follows: `LevelCelView: Modified <...>`. If you're not sure, you can drop the tag, and if the issue is most probably related to every, or many components of the app, you can use `Everywhere:` tag.
+* Create only one issue per bug/implementation request
+* Thoroughly describe the bug that you witnessed, if possible also add screenshots - there is no general pattern or template on how the issue should look like, so try to give the next person who will take care of it as many hints as possible
+* Include your platform name that you've witnessed the bug on, and libraries versions that you're using on your system
+* Do not create issues regarding build issues or other support request, if CI builds the application during pull request - the problem is most probably on your side. Please ask for help on `#request-support` channel on our Discord.
```
Comment (AJenbo) [inline CONTRIBUTING.md]: [discussion](https://github.com/diasurgical/d1-graphics-tool/discussions) on github would also be ok

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+## Creating a PR - guidelines
+
+Those guidelines won't be neccessarily as strictly executed as they should be, but it should be your responsibility to mark all the checks. If you're a future maintainer, or a future frequent contributor, you'd not want to debug a change that's not as descriptive or readable as it should be, at the same time not knowing if it introduced something important or not. So it's better for everyone to stick to the guidelines and always check them before opening a new Pull Request.
```
Comment (AJenbo) [inline CONTRIBUTING.md]:
```suggestion
Those guidelines won't be necessarily as strictly executed as they should be, but it should be your responsibility to mark all the checks. If you're a future maintainer, or a future frequent contributor, you'd not want to debug a change that's not as descriptive or readable as it should be, at the same time not knowing if it introduced something important or not. So it's better for everyone to stick to the guidelines and always check them before opening a new Pull Request.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+* Do not advertise your personal politics or religious beliefs anywhere in the code, or on our github repository. Do not add changes related to those topics.
+* Do not curse extensively in pull requests/issues
+* Do not bully someone for their github profile, bad code, profile picture, beliefs discovered on some personal page or anything that they done in the past
+* If you have questions regarding code during a review, it's good to assume someone had a reason to do something, and it's better to ask "Why is it here, shouldn't we do XYZ?" rather than "What the f... is that garbage?"
+* If your contribution is not getting attention, you can always on the `#development` channel on our Discord, maybe someone will have some time and take a look on it. Just don't expect someone to do that immediately and don't spam the link, that may annoy people.
```
Comment (AJenbo) [inline CONTRIBUTING.md]: > you can always on the `#development` channel

You can always what?

File: CONTRIBUTING.md (hunk 2 above)
Comment (tetektoza) [reply CONTRIBUTING.md]: Good catch, done :)

File: CONTRIBUTING.md (hunk 3 above)
Comment (tetektoza) [reply CONTRIBUTING.md]: Finished the sentence, done :)

File: CONTRIBUTING.md (hunk 1 above)
Comment (tetektoza) [reply CONTRIBUTING.md]: Added a reference link to the discussions tab on github, so contributors can redirect build issues here.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+* Try to prepend your issue's title with the category/component that the issue is related to, for example, if your change is mostly related to LevelCelView, you'd name your issue as follows: `LevelCelView: Modified <...>`. If you're not sure, you can drop the tag, and if the issue is most probably related to every, or many components of the app, you can use `Everywhere:` tag.
+* Create only one issue per bug/implementation request
+* Thoroughly describe the bug that you witnessed, if possible also add screenshots - there is no general pattern or template on how the issue should look like, so try to give the next person who will take care of it as many hints as possible
+* Include your platform name that you've witnessed the bug on, and libraries versions that you're using on your system
+* Do not create issues regarding build issues or other support request, if CI builds the application during pull request - the problem is most probably on your side. Please ask for help on `#request-support` channel on our Discord, or by creating a proper Q&A discussion on [discussions](https://github.com/diasurgical/d1-graphics-tool/discussions) tab on Github.
```
Comment (AJenbo) [inline CONTRIBUTING.md]:
```suggestion
* Please refrain from creating issues related to build problems or other support requests. If CI builds the application without any issues during a pull request, the problem is likely on your side. Instead, seek assistance on the #request-support channel on our Discord or initiate a proper Q&A discussion on the [Discussions](https://github.com/diasurgical/d1-graphics-tool/discussions) tab on GitHub.
```

Comment (tetektoza) [reply CONTRIBUTING.md]: Applied.

**Pull request merged**